### PR TITLE
Fixed the issue of aws: tag

### PR DESCRIPTION
**Pull request by athiruma**

### CODE REVIEW

File: cloud_governance/common/aws/utils/utils.py [hunk 1]
```diff
@@ -10,8 +11,13 @@ class Utils:
...
 
-    def __init__(self):
-        pass
+    def __init__(self, region: str = 'us-east-2'):
+        self.region = region
+        self.ec2_client = boto3.client('ec2', region_name=region)
+        self.iam_client = boto3.client('iam')
+        self.s3_client = boto3.client('s3')
+        self.elb_client = boto3.client('elb', region_name=region)
+        self.elbv2_client = boto3.client('elbv2', region_name=region)
```
Comment (ebattat) [inline cloud_governance/common/aws/utils/utils.py:21]: Why we need all the the client init here ?

File: cloud_governance/tag_cluster/remove_cluster_tags.py [hunk 2]
```diff
@@ -120,19 +134,35 @@ def get_cluster(self, clusters: list):
                                 break
         return [cluster_dict, cluster_tags]
 
-    def cluster_instance(self):
+    def cluster_update_tags(self, cluster: list, queue):
```
Comment (ebattat) [inline cloud_governance/tag_cluster/remove_cluster_tags.py:137]: queue type ?

File: cloud_governance/tag_non_cluster/remove_non_cluster_tags.py [hunk 3]
```diff
@@ -83,10 +85,10 @@ def __get_instance_tags(self, launch_time: datetime, instance_id: str, tags: lis
         user_tags = self.iam_client.get_user_tags(username=username)
         if not username:
```
Comment (ebattat) [inline cloud_governance/tag_non_cluster/remove_non_cluster_tags.py:86]: can we replace username to user ?

File: cloud_governance/tag_non_cluster/remove_non_cluster_tags.py [hunk 4]
```diff
@@ -101,17 +103,16 @@ def non_cluster_update_ec2(self, instances_list: list = None):
...
                 instance_id = item.get('InstanceId')
                 launch_time = item.get('LaunchTime')
                 add_tags = self.__get_instance_tags(launch_time=launch_time, instance_id=instance_id,
                                                     tags=item.get('Tags'))
-            if add_tags:
-                if self.dry_run == 'no':
-                    self.ec2_client.delete_tags(Resources=[instance_id], Tags=add_tags)
-                    logger.info(f'delete tags of instance: {instance_id} total: {len(add_tags)} tags: {add_tags}')
-                instances_ids.append(instance_id)
+                if add_tags:
```
Comment (ebattat) [inline cloud_governance/tag_non_cluster/remove_non_cluster_tags.py:111]: Why we have tab here ?

File: cloud_governance/tag_non_cluster/tag_non_cluster_resources.py [hunk 5]
```diff
@@ -112,11 +116,11 @@ def non_cluster_update_ec2(self, instances_list: list = None):
                 launch_time = item.get('LaunchTime')
                 add_tags = self.__get_instance_tags(launch_time=launch_time, instance_id=instance_id,
                                                     tags=item.get('Tags'))
-            if add_tags:
-                if self.dry_run == 'no':
-                    self.ec2_client.create_tags(Resources=[instance_id], Tags=add_tags)
-                    logger.info(f'added tags to instance: {instance_id} total: {len(add_tags)} tags: {add_tags}')
-                instances_ids.append(instance_id)
+                if add_tags:
```
Comment (ebattat) [inline cloud_governance/tag_non_cluster/tag_non_cluster_resources.py:119]: Why we have tab here ?

File: cloud_governance/tag_non_cluster/tag_non_cluster_resources.py [hunk 6]
```diff
@@ -274,11 +280,12 @@ def update_ami(self, images: list = None):
             search_tags.extend(self.__append_input_tags())
             if username:
                 user_tags = self.iam_client.get_user_tags(username=username)
-                tag_name = f'{username}-{image_id[-4:]}'
+                tag_name = f'{username}-{image_id[-5:]}'
```
Comment (ebattat) [inline cloud_governance/tag_non_cluster/tag_non_cluster_resources.py:283]: Please add constant for 5 with meaningful name

File: cloud_governance/tag_non_cluster/tag_non_cluster_resources.py [hunk 7]
```diff
@@ -274,11 +280,12 @@ def update_ami(self, images: list = None):
...
                 search_tags.append({'Key': 'Email', 'Value': f'{username}@redhat.com'})
                 search_tags.extend(user_tags)
             else:
-                tag_name = f'{image_id[:3]}-{self.region}-{image_id[-4:]}'
+                search_tags.append({'Key': 'User', 'Value': 'NA'})
+                tag_name = f'{image_id[:3]}-{self.region}-{image_id[-5:]}'
```
Comment (ebattat) [inline cloud_governance/tag_non_cluster/tag_non_cluster_resources.py:288]: Please add constant for 5 with meaningful name

File: cloud_governance/common/aws/utils/utils.py (hunk 1 above)
Comment (athiruma) [reply cloud_governance/common/aws/utils/utils.py:21]: to get the load_balancesrs, users,  roles, and s3. All are calling from here only.

File: cloud_governance/tag_cluster/remove_cluster_tags.py (hunk 2 above)
Comment (athiruma) [reply cloud_governance/tag_cluster/remove_cluster_tags.py:137]: It is a Multithread Queue. So I didn't know the exact type

File: cloud_governance/tag_non_cluster/tag_non_cluster_resources.py (hunk 6 above)
Comment (athiruma) [reply cloud_governance/tag_non_cluster/tag_non_cluster_resources.py:283]: ok

File: cloud_governance/tag_non_cluster/remove_non_cluster_tags.py (hunk 4 above)
Comment (athiruma) [reply cloud_governance/tag_non_cluster/remove_non_cluster_tags.py:111]: I didn't see any tab there.

File: cloud_governance/tag_non_cluster/tag_non_cluster_resources.py (hunk 6 above)
Comment (athiruma) [reply cloud_governance/tag_non_cluster/tag_non_cluster_resources.py:283]: LENGHT ??

File: cloud_governance/common/aws/utils/utils.py (hunk 1 above)
Comment (ebattat) [reply cloud_governance/common/aws/utils/utils.py:21]: Ok got it

File: cloud_governance/tag_non_cluster/remove_non_cluster_tags.py (hunk 3 above)
Comment (ebattat) [reply cloud_governance/tag_non_cluster/remove_non_cluster_tags.py:86]: Ok got it

File: cloud_governance/common/aws/utils/utils.py (hunk 1 above)
Comment (athiruma) [reply cloud_governance/common/aws/utils/utils.py:21]: Sorry, I forgot to delete it here.

File: cloud_governance/tag_non_cluster/tag_non_cluster_resources.py (hunk 5 above)
Comment (ebattat) [reply cloud_governance/tag_non_cluster/tag_non_cluster_resources.py:119]: Ok

File: cloud_governance/tag_non_cluster/tag_non_cluster_resources.py (hunk 6 above)
Comment (ebattat) [reply cloud_governance/tag_non_cluster/tag_non_cluster_resources.py:283]: SHORT_ID

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**